cmake/extensions.cmake

```diff
@@ -253,6 +254,12 @@ Please provide one of following: CONF_FILES")
     )
     list(APPEND filename_list ${PM_FILE_PREFIX}_${filename})
 
+    # Prepare search for pm_static_board.yml
+    zephyr_build_string(filename
+                        BOARD ${ZEPHYR_FILE_BOARD}
+    )
+    list(APPEND filename_list ${PM_FILE_PREFIX}_${filename})
```

Comment:
please note, this is deviating from upstream pattern of looking up files here:
https://github.com/zephyrproject-rtos/zephyr/blob/23f057804285c26806863daed1d095dd06ebc2dc/cmake/modules/extensions.cmake#L2232-L2242

The pure board version upstream is only considered when build type is not set.

At this point in time with sysbuild being introduced and system devicetree appearing in the horizon then i'm not convinced we should extend the lookup scheme for static pm configurations.

Instead, cause I see you want to allow a board specific pm static in the board dir even when build type is used (https://github.com/nrfconnect/sdk-nrf/pull/9288/commits/f74cf273ec05cba8057441415cc470f04ee90667), then you should update this file instead:
https://github.com/nrfconnect/sdk-nrf/blob/cfdb467363948550f04daa926ed5955e101441b3/cmake/partition_manager.cmake#L38-L42

with an additional:
```
ncs_file(CONF_FILES ${BOARD_DIR}
         PM board_dir_pm_static
         DOMAIN ${DOMAIN}
)
```
so that it is only a fallback for the board dir, and not in general.

Remember to adjust local var names etc so the functionality is as intended.

Reply:
Thanks for your suggestion. Changed.